### PR TITLE
Add ROS service for executing shell commands

### PR DESCRIPTION
**Pull request by okalachev**
Shortcomings for now:

* `stderr` is not captured
* timeout is not implemented

### CODE REVIEW

**Comment by okalachev** [comment]
Also thinking about renaming `output` to something else.

**Comment by sfalexrog** [comment]
*sigh*
Where do I even begin?

Okay, let's start with the API.
Almost every language (or should I say, every *real* language?) has a way to start a subprocess. There's also typically a way to provide something for standard input (typically a string) and capture output from both stdout and stderr. Some APIs even allow you to spawn a shell and execute your commands there - this might be somewhat unorthodox, but it has some uses. Out of those three things, your API does only one. Now, you might say that this API is limited by design, but why not limit it further and do *just what you need*?

And that brings me to another point: what is the purpose of this?
Why do a weird, incomplete API that will be used in exactly two cases: getting a version and running selfcheck? And it won't even be good at that, because selfcheck will time out and you'll have to fall back to reading `/rosout`. I'd rather see selfcheck split into two parts: a "backend" that would post all of the diagnostic info in a topic and a "frontend" that would just read the report.

In short, I strongly believe this is both over-engineered and underdeveloped. I am against such an interface and in favor of smaller, simplier services. Want a version? Write a service that will only post a version. Even better, write a service that would post diagnostics and read version from there. We already have at least two ways of getting a shell, do we *really* need a third, inferior one?

**Comment by okalachev** [comment]
Absolutely not, I never said this is only for getting the version and running selfcheck. Instead, this API is for doing all the stuff in web-based apps for Clover, that cannot be done another way.

Off course you can write a service for every little task and have 1000 nodes running, but you can have simpler solution that covers a lot of cases without the need of writing a custom think.

Consider this like the MAVLink interface  for encapsulating data to the autopilot shell (`SERIAL_CONTROL` message).

**Comment by okalachev** [comment]
Sending stdin is possible with `echo foo | cmd`.
